fw/tls.c

```diff
@@ -282,11 +282,8 @@ tfw_tls_encrypt(struct sock *sk, struct sk_buff *skb, unsigned int mss_now,
 	head_sz = ttls_payload_off(xfrm);
 	len = skb->len;
 	type = skb_tfw_tls_type(skb);
-	if (!type) {
-		T_WARN("%s: bad skb type %u\n", __func__, type);
-		r = -EINVAL;
-		goto err_epilogue;
-	}
+	/* Checked early before call this function. */
+	BUG_ON(!type);
```

Comment:
Is a crash inevitable in this case? If not and we still can not to crash, then let's just put WARN_ON_ONCE(). By the way, checkpatch.pl also claims unnecessary BUG statements